features/lwipstack/lwip/src/apps/altcp_tls/altcp_tls_mbedtls.c

```diff
@@ -714,12 +728,25 @@ altcp_tls_create_config(int is_server, int have_cert, int have_pkey, int have_ca
 
   mbedtls_ssl_config_init(&conf->conf);
   mbedtls_entropy_init(&conf->entropy);
-  mbedtls_ctr_drbg_init(&conf->ctr_drbg);
+#if defined(MBEDTLS_CTR_DRBG_C)
+  mbedtls_ctr_drbg_init(&conf->_drbg);
+#elif defined(MBEDTLS_HMAC_DRBG_C)
+  mbedtls_hmac_drbg_init(&conf->_drbg);
+#endif
 
   /* Seed the RNG */
-  ret = mbedtls_ctr_drbg_seed(&conf->ctr_drbg, ALTCP_MBEDTLS_RNG_FN, &conf->entropy, ALTCP_MBEDTLS_ENTROPY_PTR, ALTCP_MBEDTLS_ENTROPY_LEN);
+#if defined(MBEDTLS_CTR_DRBG_C)
+  ret = mbedtls_ctr_drbg_seed(&conf->_drbg, ALTCP_MBEDTLS_RNG_FN, &conf->entropy, ALTCP_MBEDTLS_ENTROPY_PTR, ALTCP_MBEDTLS_ENTROPY_LEN);
+#elif defined(MBEDTLS_HMAC_DRBG_C)
+  ret = mbedtls_hmac_drbg_seed(&conf->_drbg, ALTCP_MBEDTLS_RNG_FN, &conf->entropy, ALTCP_MBEDTLS_ENTROPY_PTR, ALTCP_MBEDTLS_ENTROPY_LEN);
+#endif
   if (ret != 0) {
+#if defined(MBEDTLS_CTR_DRBG_C)
     LWIP_DEBUGF(ALTCP_MBEDTLS_DEBUG, ("mbedtls_ctr_drbg_seed failed: %d\n", ret));
+#elif defined(MBEDTLS_HMAC_DRBG_C)
+    LWIP_DEBUGF(ALTCP_MBEDTLS_DEBUG, ("mbedtls_hmac_drbg_seed failed: %d\n", ret));
+#endif
```

Comment:
No, we don't want to pollute code with dead code

Reply:
So only one of those 2 conditions is always guaranteed? If that is the case then the #elif should be a #else ? Otherwise you have unexpected code execution....

Reply:
Same point as [above](https://github.com/ARMmbed/mbed-os/pull/12479#discussion_r400106285).

Reply:
Fixed